crates/essential-rest-client/src/builder_client.rs

```diff
@@ -79,6 +79,25 @@ impl EssentialBuilderClient {
         Ok(output)
     }
 
+    /// Register program
+    ///
+    /// Creates a solution to the program registry predicate and submits it.
+    pub async fn register_program(
+        &self,
+        big_bang: &BigBang,
```

Comment:
Maybe this function should expect only the program registry address. `deploy_program` does not need to expect the entire `BigBang` either. Although, because it's a ref to `BigBang`, not sure if it's better or worse to be more granular in the params to these two functions. What do you think @mitchmindtree?

Reply:
I'm happy to go either way here.

Reply:
Good spot! I agree it's probably better to only pass in the necessary address rather than the whole config